File: c/container_with_most_water.c
int maxArea(int *height, int heightSize) {
	int i = 0, j = heightSize - 1, r = 0;
	while (i < j) {
		int a;
		if (height[i] > height[j]) {
			a = (j - i) * height[j];
			j--;
		} else {
			a = (j - i) * height[i];
			i++;
		}
		if (a > r)
			r = a;
	}
	return r;
}

File: go/container_with_most_water.go
package main

func maxArea(height []int) int {
	a := 0
	l := 0
	r := len(height) - 1
	for l < r {
		if height[l] > height[r] {
			a = max(a, (r-l)*height[r])
			r--
		} else {
			a = max(a, (r-l)*height[l])
			l++
		}
	}
	return a
}
